compute/client_library/ingredients/instances/change_machine_type.py

```diff
@@ -0,0 +1,53 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+# This is an ingredient file. It is not meant to be run directly. Check the samples/snippets
+# folder for complete code samples that are ready to be used.
+# Disabling flake8 for the ingredients file, as it would fail F821 - undefined name check.
+# flake8: noqa
+
+from typing import NoReturn
+
+from google.cloud import compute_v1
+
+
+# <INGREDIENT change_machine_type>
+def change_machine_type(project_id: str, zone: str, instance_name: str, new_machine_type: str) -> NoReturn:
+    """
+    Changes the machine type of VM. The VM needs to be in the 'TERMINATED' state for this operation to be successful.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you want to use.
+        zone: name of the zone your instance belongs to.
+        instance_name: name of the VM you want to modify.
+        new_machine_type: the new machine type you want to use for the VM.
+            For example: `e2-standard-8`, `e2-custom-4-2048` or `m1-ultramem-40`
+            More about machine types: https://cloud.google.com/compute/docs/machine-resource
+    """
+    client = compute_v1.InstancesClient()
+    instance = client.get(project=project_id, zone=zone, instance=instance_name)
+
+    if instance.status != compute_v1.Instance.Status.TERMINATED.name:
+        raise RuntimeError(f"Only machines in TERMINATED state can have their machine type changed. "
```

Comment:
nit: perhaps it can be `ValueError`? But `RuntimeError` is fine as well.

Reply:
I think RuntimeError is more suitable. 

RuntimeError - there's something wrong outside of the code - some condition is not met to do something.
ValueError - someone screwed up some code passing wrong type to a method i.e. `int('test')` gives ValueError.

Reply:
FYI https://docs.python.org/3/library/exceptions.html#ValueError:
```
exception ValueError
Raised when an operation or function receives an argument that has the right type but an inappropriate value, and the situation is not described by a more precise exception such as [IndexError](https://docs.python.org/3/library/exceptions.html#IndexError).
```
But, since this is more of a "status" than value RE is fine.